test/hotspot/jtreg/gc/stringdedup/TestStringDeduplicationTools.java

```diff
@@ -410,6 +405,15 @@ public static void main(String[] args) {
             System.out.println("End: InternedTest");
         }
 
+        private static void checkNotDeduplicated(Object value1, Object value2) {
+            // Note that the following check is invalid since a concurrent GC
```

Comment:
I think the word "concurrent" can be dropped - it seems like any GC could trip over the problem.
It's just that the STW GCs are unlikely to occur and trip this because the various forceDeduplication
calls force GCs at those points, making automatically triggered GCs elsewhere unlikely.